### ISSUE TITLE
The label-studio-ml package is shipped with an unnecessary scikit-learn dependency

### ISSUE THREAD

**Issue by ludelafo** [heartexlabs/label-studio-ml-backend#131]
My understanding of the Python ecosystem is quite limited, please excuse my shortcomings related to this issue.

# Description

The [`label-studio-ml`](https://pypi.org/project/label-studio-ml/) package is shipped with `scikit-learn==0.24.1`.

This means that projects using `scikit-learn>=0.24.1` cannot be used in conjunction with `label-studio-ml==1.0.7` as it will uninstall the newer version of `scikit-learn`.

More precisely, my model uses the method [`get_feature_names_out`](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html#sklearn.feature_extraction.text.CountVectorizer.get_feature_names_out) on a [`CountVectorizer`](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html) object, [added in `scikit-learn==1.0.0`](https://scikit-learn.org/stable/whats_new/v1.0.html#changes-1-0) (please note that the method [`get_feature_names`](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html#sklearn.feature_extraction.text.CountVectorizer.get_feature_names) available in `scikit-learn==0.24.1` has been deprecated in `scikit-learn==1.0.0` as well).

```py
bag_of_words = CountVectorizer(
    stop_words="english", max_features=max_features, ngram_range=(1, ngrams)
)

bag_of_words.fit(train_words)
train_words_binary_matrix = bag_of_words.transform(train_words)
feature_names = bag_of_words.get_feature_names_out()
tfidf = TfidfTransformer(smooth_idf=False)
tfidf.fit(train_words_binary_matrix)
train_words_tfidf_matrix = tfidf.transform(train_words_binary_matrix)
```

# Steps to reproduce

```bash
# Generate the virtual environment
$ python3 -m venv .venv

# Activate the virtual environment
$ source .venv/bin/activate

# Display the current requirements
$ cat requirements.txt
scikit-learn==1.1.1

# Install the requirements
$ pip install --requirement requirements.txt
Collecting scikit-learn==1.1.1
  Using cached scikit_learn-1.1.1-cp39-cp39-macosx_10_13_x86_64.whl (8.6 MB)
Collecting joblib>=1.0.0
  Using cached joblib-1.1.0-py2.py3-none-any.whl (306 kB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting numpy>=1.17.3
  Using cached numpy-1.22.4-cp39-cp39-macosx_10_15_x86_64.whl (17.7 MB)
Collecting scipy>=1.3.2
  Using cached scipy-1.8.1-cp39-cp39-macosx_12_0_universal2.macosx_10_9_x86_64.whl (55.6 MB)
Installing collected packages: threadpoolctl, numpy, joblib, scipy, scikit-learn
Successfully installed joblib-1.1.0 numpy-1.22.4 scikit-learn-1.1.1 scipy-1.8.1 threadpoolctl-3.1.0

# Install label-studio-ml package
$ pip install label-studio-ml
Collecting label-studio-ml
  Using cached label_studio_ml-1.0.7-py3-none-any.whl (26 kB)
Collecting label-studio-tools==0.0.0.dev11
  Using cached label_studio_tools-0.0.0.dev11-py3-none-any.whl (9.8 kB)
Collecting Flask==1.1.2
  Using cached Flask-1.1.2-py2.py3-none-any.whl (94 kB)
Collecting requests<3,>=2.22.0
  Using cached requests-2.28.0-py3-none-any.whl (62 kB)
Collecting colorama>=0.4.4
  Using cached colorama-0.4.5-py2.py3-none-any.whl (16 kB)
Collecting itsdangerous==2.0.1
  Using cached itsdangerous-2.0.1-py3-none-any.whl (18 kB)
Collecting lxml>=4.2.5
  Using cached lxml-4.9.0-cp39-cp39-macosx_10_15_x86_64.whl (4.6 MB)
Collecting attrs>=19.2.0
  Using cached attrs-21.4.0-py2.py3-none-any.whl (60 kB)
Collecting scikit-learn==0.24.1
  Using cached scikit_learn-0.24.1-cp39-cp39-macosx_10_13_x86_64.whl (7.3 MB)
Collecting Pillow
  Using cached Pillow-9.1.1-cp39-cp39-macosx_10_10_x86_64.whl (3.1 MB)
Collecting attr==0.3.1
  Using cached attr-0.3.1.tar.gz (1.7 kB)
  Preparing metadata (setup.py) ... done
Collecting werkzeug==2.0.2
  Using cached Werkzeug-2.0.2-py3-none-any.whl (288 kB)
Collecting appdirs>=1.4.3
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting Jinja2==3.0.3
  Using cached Jinja2-3.0.3-py3-none-any.whl (133 kB)
Collecting click>=5.1
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1-cp39-cp39-macosx_10_9_x86_64.whl (13 kB)
Requirement already satisfied: threadpoolctl>=2.0.0 in ./.venv/lib/python3.9/site-packages (from scikit-learn==0.24.1->label-studio-ml) (3.1.0)
Requirement already satisfied: joblib>=0.11 in ./.venv/lib/python3.9/site-packages (from scikit-learn==0.24.1->label-studio-ml) (1.1.0)
Requirement already satisfied: scipy>=0.19.1 in ./.venv/lib/python3.9/site-packages (from scikit-learn==0.24.1->label-studio-ml) (1.8.1)
Requirement already satisfied: numpy>=1.13.3 in ./.venv/lib/python3.9/site-packages (from scikit-learn==0.24.1->label-studio-ml) (1.22.4)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.6.15-py3-none-any.whl (160 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Using legacy 'setup.py install' for attr, since package 'wheel' is not installed.
Installing collected packages: attr, appdirs, werkzeug, urllib3, Pillow, MarkupSafe, lxml, itsdangerous, idna, colorama, click, charset-normalizer, certifi, attrs, scikit-learn, requests, label-studio-tools, Jinja2, Flask, label-studio-ml
  Running setup.py install for attr ... done
  Attempting uninstall: scikit-learn
    Found existing installation: scikit-learn 1.1.1
    Uninstalling scikit-learn-1.1.1:
      Successfully uninstalled scikit-learn-1.1.1
Successfully installed Flask-1.1.2 Jinja2-3.0.3 MarkupSafe-2.1.1 Pillow-9.1.1 appdirs-1.4.4 attr-0.3.1 attrs-21.4.0 certifi-2022.6.15 charset-normalizer-2.0.12 click-8.1.3 colorama-0.4.5 idna-3.3 itsdangerous-2.0.1 label-studio-ml-1.0.7 label-studio-tools-0.0.0.dev11 lxml-4.9.0 requests-2.28.0 scikit-learn-0.24.1 urllib3-1.26.9 werkzeug-2.0.2
```

# What is my understanding

I downloaded and extracted the [`label-studio-ml-1.0.7.tar.gz`](https://pypi.org/project/label-studio-ml/#files) file and searched for the usage of `scikit-learn` or `sklearn` and no results were found, except a mention in the `label_studio_ml.egg-info/requires.txt` file, where we can find the `scikit-learn==0.24.1` dependency. It seems to indicate that the `scikit-learn` is not necessary to the `label-studio-ml` dependency.

# Related issues/pull requests

I have found the following issues/pull requests related to this issue:

- #107

# How can I contribute

Please let me know how can I contribute to this repository and fix this issue; I would happily do it.

**Comment by KonstantinKorotaev** [comment]
Hi @ludelafo 
I have released ML backend version 1.0.8 without scikit-learn dependency, you can use for your task.
